src/visualization/components/internal/DecimalPlaces.tsx

```diff
@@ -0,0 +1,79 @@
+// Libraries
+import React, {FC, useState} from 'react'
+
+// Components
+import {
+  AutoInput,
+  AutoInputMode,
+  Form,
+  Input,
+  InputType,
+} from '@influxdata/clockface'
+
+// Utils
+import {convertUserInputToNumOrNaN} from 'src/shared/utils/convertUserInput'
+
+// Constants
+import {
+  MIN_DECIMAL_PLACES,
+  MAX_DECIMAL_PLACES,
+} from 'src/visualization/constants'
+
+interface DecimalPlacesProps {
+  isEnforced: boolean
+  digits: number
+  update: (obj: any) => void
+}
+
+export const DecimalPlaces: FC<DecimalPlacesProps> = (
+  props: DecimalPlacesProps
+) => {
+  const {isEnforced, digits, update} = props
+  const [decimalPlaces, setDecimalPlaces] = useState<number | null>(digits)
+
+  const setDigits = (updatedDigits: number | null) => {
+    setDecimalPlaces(updatedDigits)
+    if (Number.isNaN(updatedDigits) === false) {
```

Comment:
Just so I'm clear, this is updating the state before the check if it's a valid digit? Why is that the behavior rather than only updating the state if `updatedDigits` isn't `NaN`.

Also, is the explicit check for `false` for readability purposes, or will `if (!Number.isNaN(updatedDigits))` produce different behavior?

Reply:
This change prevents the visualization from crashing. Previously, immediately updating the Redux state will have the NaN or null sent back in as a prop which gets sent to Giraffe, which crashes the viz.

By having a local state, we can highlight the input (with an error border) to show the user that the number of decimal places should not be blank. (I don't think they can type anything other than a number in this particular input). And while the input is in this error state, do not update the Redux state just yet. Wait till the input is valid before updating the Redux state.

Local state saves the day by handling the bad input and only filtering the good input to Redux state, in a sense.